app/src/androidTest/java/org/mozilla/fenix/ui/DownloadTest.kt

```diff
@@ -50,49 +61,288 @@ class DownloadTest {
+            verifyDownloadedFileIcon()
+        }
+    }
+
+    @SmokeTest
+    @Test
+    fun downloadTxtTypeTest() {
+        fileName = "textfile.txt"
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(testPage.toUri()) {
+        }.clickDownloadLink(fileName) {
+            verifyDownloadPrompt(fileName)
+        }.clickDownload {
+            verifyDownloadNotificationPopup()
+        }.closePrompt {
+        }.openThreeDotMenu {
+        }.openDownloadsManager {
+            waitForDownloadsListToExist()
+            verifyDownloadedFileName(fileName)
+            verifyDownloadedFileIcon()
+        }
+    }
+
+    @SmokeTest
+    @Test
+    fun downloadPNGTypeTest() {
+        fileName = "web_icon.png"
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(testPage.toUri()) {
+        }.clickDownloadLink(fileName) {
+            verifyDownloadPrompt(fileName)
+        }.clickDownload {
+            verifyDownloadNotificationPopup()
+        }.closePrompt {
+        }.openThreeDotMenu {
+        }.openDownloadsManager {
+            waitForDownloadsListToExist()
+            verifyDownloadedFileName(fileName)
+            verifyDownloadedFileIcon()
+        }
+    }
+
+    @SmokeTest
+    @Test
+    fun downloadWebmTypeTest() {
+        fileName = "videoSample.webm"
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(testPage.toUri()) {
+        }.clickDownloadLink(fileName) {
+            verifyDownloadPrompt(fileName)
+        }.clickDownload {
+            verifyDownloadNotificationPopup()
+        }.closePrompt {
+        }.openThreeDotMenu {
+        }.openDownloadsManager {
+            waitForDownloadsListToExist()
+            verifyDownloadedFileName(fileName)
+            verifyDownloadedFileIcon()
+        }
+    }
+
+    @SmokeTest
+    @Test
+    fun downloadCSVTypeTest() {
+        fileName = "CSVfile.csv"
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(testPage.toUri()) {
+        }.clickDownloadLink(fileName) {
+            verifyDownloadPrompt(fileName)
+        }.clickDownload {
+            verifyDownloadNotificationPopup()
+        }.closePrompt {
+        }.openThreeDotMenu {
+        }.openDownloadsManager {
+            waitForDownloadsListToExist()
+            verifyDownloadedFileName(fileName)
+            verifyDownloadedFileIcon()
+        }
+    }
+
+    @SmokeTest
+    @Test
+    fun downloadXMLTypeTest() {
+        fileName = "XMLfile.xml"
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(testPage.toUri()) {
+        }.clickDownloadLink(fileName) {
+            verifyDownloadPrompt(fileName)
+        }.clickDownload {
+            verifyDownloadNotificationPopup()
+        }.closePrompt {
+        }.openThreeDotMenu {
+        }.openDownloadsManager {
+            waitForDownloadsListToExist()
+            verifyDownloadedFileName(fileName)
+            verifyDownloadedFileIcon()
+        }
```

Comment:
A lot of the above tests are duplicate tests. I wonder if it would better to just loop through file types in a list and throw an exception if one type fails and continue with other types. Might be a testing-standard related conversation that Richard might have an opinion on.

Reply:
on it

Reply:
I had to use JUnit's Parameterized class which runs every test in that class with the data provided. So I needed to move the test into its own class for that.